proxy/jdbc/src/main/scala/io/cloudstate/proxy/jdbc/SlickEnsureTablesExistReadyCheck.scala

```diff
@@ -16,61 +16,76 @@
+  private val autoCreateTables = proxyConfig.getBoolean("jdbc.auto-create-tables")
+
+  // Get a hold of the akka-jdbc slick database instance
+  private val eventSourcedSlickDatabase =
+    SlickExtension(system).database(ConfigFactory.parseMap(Map(ConfigKeys.useSharedDb -> "slick").asJava))
+
+  // Get a hold of the cloudstate.proxy.value-entity-persistence-store.jdbc.database.slick database instance
+  private val valueEntitySlickDatabase = JdbcSlickDatabase(proxyConfig)
 
   private val check: () => Future[Boolean] = if (autoCreateTables) {
-    // Get a hold of the akka-jdbc slick database instance
-    val db = SlickExtension(system).database(ConfigFactory.parseMap(Map(ConfigKeys.useSharedDb -> "slick").asJava))
-
-    val actor = system.actorOf(
-      BackoffSupervisor.props(
-        BackoffOpts.onFailure(
-          childProps = Props(new EnsureTablesExistsActor(db)),
-          childName = "jdbc-table-creator",
-          minBackoff = 3.seconds,
-          maxBackoff = 30.seconds,
-          randomFactor = 0.2
+    tableCreateCombinations() match {
+      case Nil => () => Future.successful(true)
+      case combinations =>
+        val actor = system.actorOf(
+          BackoffSupervisor.props(
+            BackoffOpts.onFailure(
+              childProps = Props(new EnsureTablesExistsActor(combinations)),
+              childName = "jdbc-table-creator",
+              minBackoff = 3.seconds,
+              maxBackoff = 30.seconds,
+              randomFactor = 0.2
+            )
+          ),
+          "jdbc-table-creator-supervisor"
         )
-      ),
-      "jdbc-table-creator-supervisor"
-    )
 
-    implicit val timeout = Timeout(10.seconds) // TODO make configurable?
-    import akka.pattern.ask
+        implicit val timeout = Timeout(proxyConfig.getDuration("jdbc.create-tables-timeout").toMillis.millis)
+        import akka.pattern.ask
 
-    () => (actor ? EnsureTablesExistsActor.Ready).mapTo[Boolean]
+        () => (actor ? EnsureTablesExistsActor.Ready).mapTo[Boolean]
+    }
   } else { () =>
     Future.successful(true)
   }
 
   override def apply(): Future[Boolean] = check()
+
+  private def tableCreateCombinations(): Seq[SlickCreateTables] = {
+    val config = system.settings.config
+    val eventSourcedEnabled = config.getBoolean("cloudstate.proxy.journal-enabled")
+    val valueEntityEnabled = config.getBoolean("cloudstate.proxy.value-entity-enabled")
+    (eventSourcedEnabled, valueEntityEnabled) match {
+      case (true, true) =>
+        Seq(
+          new EventSourcedSlickCreateTable(system, eventSourcedSlickDatabase),
+          new ValueEntitySlickCreateTable(system, valueEntitySlickDatabase)
+        )
+      case (true, false) =>
+        Seq(new EventSourcedSlickCreateTable(system, eventSourcedSlickDatabase))
+      case (false, true) =>
+        Seq(new ValueEntitySlickCreateTable(system, valueEntitySlickDatabase))
+      case (false, false) => Seq()
```

Comment:
Rather than matching the various combinations of enabled/disabled, can just join two seqs, like:

```scala
(if (eventSourcedEnabled) Seq(new EventSourcedSlickCreateTable(system, eventSourcedSlickDatabase)) else Seq.empty) ++
(if (valueEntityEnabled) Seq(new ValueEntitySlickCreateTable(system, valueEntitySlickDatabase)) else Seq.empty)
```